rfcs/accordians-component.md

```diff
@@ -0,0 +1,141 @@
+# Accordians Component
+
+## Usage
+
+```js
+import {Accordians, Root, Toggle} from './index';
+
+export default () => {
+  return (
+    <div>
+      <Accordians
+        accordians={[
+          {
+            title: 'First title',
+            content: <div>First content</div>
+          },
+          {
+            title: 'Second title',
+            content: <div>Second content</div>
+          }
+        ]}
+      />
+    </div>
```

Comment:
What do you think about an API more like this:

```
<Accordion exclusive>
  <Panel title="This is panel header 1" key="1">
    <p>{text}</p>
  </Panel>
  <Panel title="This is panel header 2" key="2">
    <p>{text}</p>
  </Panel>
  <Panel title="This is panel header 3" key="3">
    <p>{text}</p>
  </Panel>
</Accordion>
```

Kind of similar to RadioGroup – accordion would communicate which panel is open through react context. The `exclusive: boolean` prop makes it so that only one panel can be open at once. Just feels little cleaner and more react-like than passing an array with react nodes inside of it.

@abmai @nadiia thoughts?

Reply:
I thought about it. But for Select we chose to provide it via attribute. For instance because content will not render by default. However, in this case we have to introduce one more abstract `Panel` and literally extract `title` to put it somewhere else, as title doesn't belong to content part. For me it looks as over-complication.

Reply:
> But for Select we chose to provide it via attribute. For instance because content will not render by default.

We are passing it via attribute for Select, but there are some slight differences. Select could have hundreds or thousands of items, Accordion would probably have fewer. Also the select items are just POJOs, in this case you have react elements in the array.

> literally extract title to put it somewhere else

Assuming the markup for Panel looks like below:

```
  <Panel title="This is panel header 3" key="3">
    <p>{text}</p>
  </Panel>
```

It seems pretty trivial to map this to the following markup inside `Panel`

```js
<StyledPanel>
  <StyledPanelTitle>{title}</StyledPanelTitle>
  {isOpen && <StyledPanelContent>{children}</StyledPanelContent>}
</StyledPanel>
```

Though maybe I'm missing something.

Reply:
Context certainly would complicate things though, I agree with that. `overrides` also becomes more tricky when we introduce another `Panel` component. I think there are trade-offs with both designs. Curious to get others opinions here.

Reply:
Select items are not so POJO, they can even accept any React markup.
Although amount of items of accordion is small, but each of those can contain a huge piece of markup. For instance I made checkout flow with accordions, and each item contained a full experience of forms, cms content, etc.

Reply:
updated with items as children. Called `AccordionItem` to not mix with real `Panel` components